homeworks/09-chokidar/src/dirWatcher.ts

```diff
@@ -0,0 +1,52 @@
+const fs = require("fs");
+const path = require("path");
+const { EventEmitter } = require("events");
+
+class DirWatcher {
+  dirPath: string;
+  delay: number;
+  maxFiles: number | null;
+  eventEmitter: typeof EventEmitter;
+  cache = new Map<string, string>();
+
+  constructor(
+    {
+      delay,
+      pathToDir,
+      maxFiles,
+    }: { delay: number; pathToDir: string; maxFiles: number | null },
+    eventEmitter: typeof EventEmitter
+  ) {
+    this.dirPath = pathToDir;
+    this.delay = delay;
+    this.maxFiles = maxFiles;
+    this.eventEmitter = eventEmitter;
+  }
+
+  watch() {
+    setInterval(async () => {
+      await fs.readdir(this.dirPath, (error: Error, files: string[]) => {
```

Comment:
There should be no `await` here because `fs.readdir` is not an async method.

Also `error` has wrong type, remove the type so it is inferred automatically.